### ISSUE TITLE
Resolving Filter Paths

### ISSUE THREAD

**Issue by iandol** [htdebeer/pandocomatic#37]
With a filter `removeHR` in `DATA-DIR/filters` with Pandoc we can do this:

```
pandoc -F removeHR -o Lu.docx 'Lu.md'
```

And pandoc will search `DATA-DIR/filters` to use the filter. However if I use no path in pandocomatic.yaml, pandocomatic rewrites the root data-dir and pandoc fails:

```yaml
#######################################
  docx-with-refs:
    pandoc:
      filter: 
        - 'removeHR' #remove horizontal rules
```

```
Error running pandoc => error while running:

pandoc	--standalone \
	--filter=/Users/ian/.pandoc/removeHR \
	--bibliography=/Users/ian/.pandoc/Core.json \
	--csl=/Users/ian/.pandoc/csl/neuron.csl \
	--from=markdown \
	--to=docx \
	--reference-doc=custom.docx \
	--dpi=300 \
	--output=./Luu.docx

Pandoc responded with:

Error running filter /Users/ian/.pandoc/removeHR:
Could not find executable /Users/ian/.pandoc/removeHR
```

You can see pandocomatic has rewritten to path relative to the root data-dir `/Users/ian/.pandoc/removeHR` which fails. Why not just let Pandoc handle the path resolution? If I specify an absolute path then this works fine (which is what I've always done, but it would be nice to not need to specify absolute paths)...

(error resolution working nicely BTW!)

**Comment by htdebeer** [comment]
Thanks!

The path resolution happens in the method [Configuration#update_path](https://github.com/htdebeer/pandocomatic/blob/8021e213f3181b03de14c00c5d8159f77ddddf52/lib/pandocomatic/configuration.rb#L258). There are three cases:

1. Starts with `./`, which points to a file local relative to the src-dir
2. Starts with `/`, which points to a absolute path on the file system
3. Starts not with a `./` or a `/`, which points to a path relative to the data-dir.

I think I can add a extra resolve step in case 3 when pandocomatic cannot find the file in the data-dir: it should look for it in pandoc's data-dir.

I am wondering if I should automatically look in the `filters` subdirectory or if the user should refer to filters in that directory by `filters/my-little-filter.exe`?

**Comment by iandol** [comment]
Pandoc has its own "special-case" [path resolution for filters](http://pandoc.org/MANUAL.html#reader-options):

> In order of preference, pandoc will look for filters in
    1. a specified full or relative path (executable or non-executable)
    2. $DATADIR/filters (executable or non-executable) 
    3. $PATH (executable only)

And templates are special-case searched in `$DATADIR/templates/`. I suppose your path resolution mechanism (3) is flexible across CSS/Filters/Templates without any special-case exceptions. In which case I suppose defaulting to the `DATADIR` is the most useful. 

I think now I understand the path resolution mechanism (and that I don't need to use absolute paths), you should perhaps leave it as is and I can just specify filters/* or css/* or templates/* and your mechanism (3) will work in all cases.

**Comment by iandol** [comment]
I tried switching to using mechanism (3) but for --reference-doc it failed:

```
👉  pandocomatic -b Lu.md
pandoc	--standalone \
	--filter=/Users/ian/.pandoc/filters/removeHR \
	--filter=/Users/ian/.pandoc/filters/authorRemoveHash \
	--bibliography=/Users/ian/.pandoc/Core.json \
	--csl=/Users/ian/.pandoc/csl/eneuro.csl \
	--from=markdown \
	--to=docx \
	--reference-doc=templates/custom.docx \
	--dpi=300 \
	--output=./Lu.docx
pandoc: templates/custom.docx: openBinaryFile: does not exist (No such file or directory)
Pandocomatic needed 3.6 seconds to convert 'Lu.md'.
```

So it seems this is not being applied there...

**Comment by htdebeer** [comment]
That is true. It is applied to all items in [`PANDOC_OPTIONS_WITH_PATH`](https://github.com/htdebeer/pandocomatic/blob/8021e213f3181b03de14c00c5d8159f77ddddf52/lib/pandocomatic/command/convert_file_command.rb#L157) and it has not been updated. There's even a TODO for it :-)

However, `reference-doc` is in that list, so it should have worked. But I see that I replace the `-` with a `_` before checking if they are in the list, while the items in the list still have the `-` in their names... Talking about a stupid bug!

I'll fix this today and create a new release.

**Comment by htdebeer** [comment]
Fix in release 0.1.4.15.

**Comment by iandol** [comment]
Great, very quick!!!! Thanks as always Huub.
